pkg/build/api/v1/types.go

```diff
@@ -54,6 +67,62 @@ type BuildSpec struct {
 	CompletionDeadlineSeconds *int64 `json:"completionDeadlineSeconds,omitempty"`
 }
 
+// BuildTriggerCause holds information about a triggered build. It is used for
+// displaying build trigger data for each build and buildconfig in oc
+// describe. It's also used to pass build trigger information from the
+// controller/event that fired the build to the generator.
+type BuildTriggerCause struct {
+	// Reason is used to store a human readable reason for why the build was
+	// triggered. E.g.: "Build manually triggered by user", "Build triggered by
+	// ConfigChange update", "Build triggered by ImageChange for image:
+	// FOO:BAR", etc.
+	Reason BuildTriggerReason `json:"reason,omitempty"`
```

Comment:
`Reason` is not a human readable field - it is machine readable.  See `api.Status`, but what you are describing is the `Message` field.